tests/warn/infer-or-type.scala

```diff
@@ -0,0 +1,8 @@
+//> using options -Winfer-union
+
+case class Pair[A](x: A, y: A)
+
+def test = {
+  val _ = List(1).contains("") // warn
```

Comment:
Maybe we could add `infer-or-type.check` to make sure we get the correct warnings in those cases and how they are printed?